compiler/src/dotty/tools/debug/ResolveReflectEval.scala

```diff
@@ -0,0 +1,346 @@
+                }
+              val value = gen.unboxIfValueClass(variable, args.head)
+              val typeSymbol = variable.info.typeSymbol
+              val elemField = typeSymbol.info.decl(termName("elem")).symbol
+              gen.setField(tree)(capture, elemField.asTerm, value)
+            case ReflectEvalStrategy.MethodCapture(variable, method, isByName) =>
+              val rawCapture = gen
+                .getMethodCapture(method, variable.name)
+                .getOrElse {
+                  report.error(s"No capture found for $variable in $method", tree.srcPos)
+                  ref(defn.Predef_undefined)
+                }
+              val capture = if isByName then gen.evaluateByName(rawCapture) else rawCapture
+              val capturedValue = gen.derefCapturedVar(capture, variable)
+              gen.boxIfValueClass(variable, capturedValue)
+            case ReflectEvalStrategy.MethodCaptureAssign(variable, method) =>
+              val capture = gen
+                .getMethodCapture(method, variable.name)
+                .getOrElse {
+                  report.error(s"No capture found for $variable in $method", tree.srcPos)
+                  ref(defn.Predef_undefined)
+                }
+              val value = gen.unboxIfValueClass(variable, args.head)
+              val typeSymbol = variable.info.typeSymbol
+              val elemField = typeSymbol.info.decl(termName("elem")).symbol
+              gen.setField(tree)(capture, elemField.asTerm, value)
+            case ReflectEvalStrategy.StaticObject(obj) => gen.getStaticObject(obj)
+            case ReflectEvalStrategy.Field(field, isByName) =>
+              // if the field is lazy, if it is private in a value class or a trait
+              // then we must call the getter method
+              val fieldValue =
+                if field.is(Lazy) || field.owner.isValueClass || field.owner.is(Trait)
+                then gen.callMethod(tree)(qualifier, field.getter.asTerm, Nil)
+                else
+                  val rawValue = gen.getField(tree)(qualifier, field)
+                  if isByName then gen.evaluateByName(rawValue) else rawValue
+              gen.boxIfValueClass(field, fieldValue)
+            case ReflectEvalStrategy.FieldAssign(field) =>
+              val arg = gen.unboxIfValueClass(field, args.head)
+              if field.owner.is(Trait) then
+                gen.callMethod(tree)(qualifier, field.setter.asTerm, List(arg))
+              else gen.setField(tree)(qualifier, field, arg)
+            case ReflectEvalStrategy.MethodCall(method) => gen.callMethod(tree)(qualifier, method, args)
+            case ReflectEvalStrategy.ConstructorCall(ctr, cls) => gen.callConstructor(tree)(qualifier, ctr, args)
+        case _ => super.transform(tree)
+
+  private def isReflectEval(symbol: Symbol)(using Context): Boolean =
+    symbol.name == termName("reflectEval") && symbol.owner == config.expressionClass
```

Comment:
Create the `termName("reflectEval")` once and for all as a field of the class.

Possibly even better, find `config.expressionClass.decl(termName("reflectEval"))` once and for all.

Reply:
Ok, I created `termName("reflectEval")` once and for all.